pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -181,38 +182,43 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 	)
 	errors = append(errors,
 		appendErrors(v1helpers.UpdateStatus(c.operatorClient, func(status *operatorv1.OperatorStatus) error {
-			resourcemerge.SetDaemonSetGeneration(&status.Generations, actualDaemonSet)
+			resourcemerge.SetDeploymentGeneration(&status.Generations, actualDeployment)
 			return nil
 		}))...,
 	)
 
-	daemonSetHasAllPodsAvailable := actualDaemonSet.Status.NumberAvailable == actualDaemonSet.Status.DesiredNumberScheduled
-	if !daemonSetHasAllPodsAvailable {
-		numNonAvailablePods := actualDaemonSet.Status.DesiredNumberScheduled - actualDaemonSet.Status.NumberAvailable
+	desiredReplicas := int32(1)
+	if actualDeployment.Spec.Replicas != nil {
+		desiredReplicas = *(actualDeployment.Spec.Replicas)
+	}
+
+	deploymentHasAllPodsAvailable := actualDeployment.Status.AvailableReplicas == desiredReplicas
```

Comment:
do we enforce that two replicasets for the deployment don't overlap and increase this number temporarily although we are still progressing?

Reply:
No, we do not enforce no overlap. As per your previous comment, health should be determined by comparing replica count to master count.